tsfile/src/main/java/org/apache/iotdb/tsfile/utils/BloomFilter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.tsfile.utils;
+
+import java.util.BitSet;
+
+public class BloomFilter {
+
+  private static final int MINIMAL_SIZE = 256;
+  private static final int MAXIMAL_HASH_FUNCTION_SIZE = 8;
+  private static final int[] seeds = new int[]{5, 7, 11, 19, 31, 37, 43, 59};
+  private int size;
+  private int hashFunctionSize;
+  private BitSet bits;
+  private HashFunction[] func;
+
+  // do not try to initialize the filter by construction method
+  private BloomFilter(byte[] bytes, int size, int hashFunctionSize) {
+    this.size = size;
+    this.hashFunctionSize = hashFunctionSize;
+    func = new HashFunction[hashFunctionSize];
+    for (int i = 0; i < hashFunctionSize; i++) {
+      func[i] = new HashFunction(size, seeds[i]);
+    }
+    bits = BitSet.valueOf(bytes);
+  }
+
+  private BloomFilter(int size, int hashFunctionSize) {
+    this.size = size;
+    this.hashFunctionSize = hashFunctionSize;
+    func = new HashFunction[hashFunctionSize];
+    for (int i = 0; i < hashFunctionSize; i++) {
+      func[i] = new HashFunction(size, seeds[i]);
+    }
+    bits = new BitSet(size);
+  }
+
+  /**
+   * get empty bloom filter
+   *
+   * @param errorPercent the tolerant percent of error of the bloom filter
+   * @param numOfString the number of string want to store in the bloom filter
+   * @return empty bloom
+   */
+  public static BloomFilter getEmptyBloomFilter(double errorPercent, int numOfString) {
+    errorPercent = Math.max(errorPercent, 0.01);
+    errorPercent = Math.min(errorPercent, 0.1);
```

Comment:
Why adding these constraints?

Reply:
There are two reasons:
1.If user set errorPercent smaller than 0 or bigger than 1, iotdb will crash.
2. If user set errorPercent very small, such as 0.00000001, the size of bloom filter will very big (GB level), so I add these constraints

Reply:
Yeah, but I still wonder why the specific range [0.01, 0.1] is used. Does any literature argue for that, or you just decide randomly?

Reply:
For 1 million time series, 0.01 error rate will use 10MB space. I think this may be too large if we specify a smaller error rate.
And then, bloom filter in hadoop's default error rate is 0.05.
So I choose 0.01 - 0.1